pytorch_lightning/utilities/migration.py

```diff
@@ -34,7 +35,12 @@ class pl_legacy_patch:
             torch.load("path/to/legacy/checkpoint.ckpt")
     """
 
+    def __init__(self) -> None:
+        # Create a lock to ensure no race condition with deleting sys modules
+        self.lock = threading.Lock()
```

Comment:
Rather protected?

Reply:
That's a good suggestion, @Borda - thanks! I've made it protected now.